### PR TITLE
ta/alloca

### PR DESCRIPTION
**Pull request by ThomasAdam**
- deprecate {Pixmap,Icon}Path
- WIP: war on alloca()

### CODE REVIEW

**Comment by ThomasAdam** [comment]
@omar-polo 

FYI -- I've started to declare war on `alloca()` -- I've not tested this, just checked it compiles.  I may well have missed some `alloca()` calls, and/pr broken functionality.

But if you wanted to use this as a starter, feel free!

File: libs/Bindings.c [hunk 1]
```diff
@@ -827,8 +827,7 @@ KeySym FvwmStringToKeysym(Display *dpy, char *key)
 	}
 	else
 	{
-		s = alloca(strlen(key) + 1);
-		strcpy(s, key);
+		snprintf(s, sizeof(key), key);
```
Comment (omar-polo) [inline libs/Bindings.c:830]: This bit reads wrong: it writes to an un-initialized pointer `s`.  I believe it should be converted to either strdup or a local buffer (i.e. declaring `s` as `char s[32]` or any other sensible maximum size.)

Comment (ThomasAdam) [reply libs/Bindings.c:830]: Oops.  You're entirely correct.  Will fix.

**Comment by ThomasAdam** [comment]
OK -- that's all the occurrences, except for some YACC thing in FvwmScript which I am not fussed about.  I've given this a quick test, and fvwm3 hasn't segfaulted yet...

**Pull request merged**